dra-evolution/README.md

```diff
@@ -118,36 +135,31 @@ spec:
       requests:
         cpu: 10m
         memory: 10Mi
-    devices:
-    - name: gpu
-  deviceClaims:
-    devices:
-    - name: gpu
-      class: example.com-foozer
+      devices:
+      - claimName: gpu
+        deviceName: gpu-one
+  - image: registry.k8s.io/pause:3.6
+    name: my-container
+    resources:
+      requests:
+        cpu: 10m
+        memory: 10Mi
+      devices:
+      - claimName: gpu
+        deviceName: gpu-two
+  resourceClaims:
+  - name: gpu
+    source:
+      resourceClaimTemplate: two-foozers
 ```
 
-Each entry in `spec.deviceClaims.devices` is just a name/class pair, but in fact
-serves as a template to generate claims that exist with the lifecycle of the
-pod. We may want to add `ObjectMeta` here as well, since it is behaving as a
-template, to allow setting labels, etc.
-
-The second form of `DeviceClaims` is a single struct with an ObjectMeta, and a
-claim name. The key with this form is that it is not *list* of named objects.
-Instead, it is a reference to a single claim object, and the named entries are
-*inside* the referenced object. This is to avoid a two-key mount in the
-`spec.containers[*].devices` entry. If that's not important, then we can tweak
-this a bit. In any case, this form allows claims which follow the lifecycle of
-the pod, similar to the first form. Since a top-level API claim spec can can
-contain multiple claim instances, this should be equally as expressive as if we
-included `matchAttributes` in the `PodSpec`, without having to do so.
-
-The third form of `DeviceClaims` is just a string; it is a claim name and allows
-the user to share a pre-provisioned claim between pods.
-
-Given that the first and second forms both have a template-like structure, we
-may want to combine them and use two-key indexing in the mounts. If we do so, we
-still want the direct specification of the class, so that the most common case
-does not need separate object just to reference a class.
+Resource classes are capable of describing everything that a user might put
+into a claim. Therefore a simple claim or claim template might contain nothing
+but a resource class name.
+
+How devices are named inside this class needs to be part of the class
```

Comment:
A class may also contain default requests. This is for the use case where an admin pre-defines a set of devices that go together:
https://github.com/kubernetes-sigs/wg-device-management/blob/c920b2be9077bdb0d0ba6369e1d6d0f89239982c/dra-evolution/testdata/classes.yaml#L183-L198

I'm okay with removing `defaultRequests`. I added it because it seemed required to completely support the "users only use classes" approach, but it looks like we are moving away from that.

Reply:
Yep. See "highly expressive classes" in https://github.com/kubernetes-sigs/wg-device-management/pull/14#discussion_r1611059303.